### PR TITLE
Make AutoTargetIgnore upgradeable

### PR DESCRIPTION
**Pull request by abcdefg30**
I know that adding more PRs to the milestone is not very helpful, sorry for that.
However I think that this is an easy fix (you could trick the soviet05 mission by selling the radar and thus making it impossible to fail the "defend the radar" objective) and not very hard to test.

Just nuke the borders of the base (not destroying the radar), and then spam the base with tesla coils. They should demolish everything other than the radar. Then sell them off and capture the radar. A small allied force will appear and attack the radar (which shows that it doesn't ignore autotargeting anymore).

### CODE REVIEW

**Comment by GraionDilach** [comment]
Would it simplify the Lua if you add a side-based prereq to the radar and then use that prereq to grant the condition? Just asking.

**Review by GraionDilach — APPROVED** [review]
:+1: regardless of the above.

**Comment by abcdefg30** [comment]
> Would it simplify the Lua if you add a side-based prereq to the radar and then use that prereq to grant the condition? Just asking.

I don't think so. Using only yaml would save us just one line of lua (`Radar.GrantCondition("captured")`). Imho that one line is simpler than a block of yaml. ^^
(The other lua changes are to further sort the in the OP mentioned bug out.)

File: mods/ra/maps/soviet-05/soviet05.lua [hunk 1]
```diff
@@ -206,14 +206,16 @@ WorldLoaded = function()
 	Trigger.OnDamaged(mcvtransport, Expand)
 
 	Trigger.OnKilled(Radar, function()
```
Comment (pchote) [inline mods/ra/maps/soviet-05/soviet05.lua]: Shouldn't this be `OnRemovedFromWorld`?  What if the player sells it?

Comment (abcdefg30) [inline mods/ra/maps/soviet-05/soviet05.lua]: At this point the Radar belongs to the AI. The situation you describe is handled [here](https://github.com/OpenRA/OpenRA/pull/12863/files#diff-28ca1ed8051b1dd53a820645c0bfac2cR232).

**Review by pchote — APPROVED** [review]
The trait changes look correct, but I haven't tested the mission.

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/431affd29de0fcd88dbd2c84a6691715d86ab836

**Pull request merged**